packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,19 +37,94 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// A position that the [FloatingActionButton] can dock to in a [Scaffold].
+/// 
+/// [getOffset] 
+class FabPosition {
+  /// End-aligned FAB, floating at the bottom of the screen.
+  /// 
+  /// This is the default alignment of FABs in Material apps.
+  static const FabPosition endFloat = const _EndFloatFab();
+
+  /// Centered FAB, floating at the bottom of the screen.
+  static const FabPosition centerFloat = const _CenterFloatFab();
+
+  const FabPosition(this.getOffset);
+
+  /// Places the FAB with the layout parameters of the [Scaffold].
+  /// 
+  /// [scaffoldSize] is the size of the whole [Scaffold].
+  /// [fabSize] is the size of the [FloatingActionButton] to be placed.
+  /// [snackBarSize] is the size of the Scaffold's [SnackBar] (if available).
+  /// [bottomSheetSize] is the size of the [BottomSheet] (if available).
+  /// [textDirection] is the direction used by the Scaffold (either RTL or LTR).
+  /// [contentBottom] is the height from the top where the Scaffold's body ends.
+  /// [horizontalPadding] is padding the Scaffold would like the FAB to observe.
```

Comment:
these will all end up in one paragraph; if that's not what you intended, consider prefixing them with ` * ` or putting a blank line between each one.

Also since these are just arguments, consider using back ticks instead of square brackets to refer to them.

Reply:
Done.